app/src/main/java/com/github/se/travelpouch/ui/dashboard/CalendarViewComposable.kt

```diff
@@ -1,39 +1,204 @@
+                    style =
+                        androidx.compose.ui.text.TextStyle(
+                            textAlign = TextAlign.Center,
+                            fontSize = 16.sp,
+                            color =
+                                if (isSelected) Color.White
+                                else Color.Black)) // TODO Use dark mode/Material theme
+
+                if (isEventDay) {
+                  Box(
+                      modifier =
+                          Modifier.size(8.dp)
+                              .align(Alignment.BottomCenter)
+                              .background(Color.Red, shape = CircleShape))
+                }
+              }
+        }
+      }
+}
+
+/**
+ * Wrapper Composable for Calendar with Back and Forward navigation arrows.
+ *
+ * @param selectedDate The currently selected date.
+ * @param events A list of dates with events.
+ * @param onDateSelected Callback when a date is selected.
+ * @param modifier Modifier to be applied to the calendar.
+ */
+@Composable
+fun CalendarView(
+    selectedDate: Date,
+    events: List<Date>,
+    onDateSelected: (Date) -> Unit,
+    modifier: Modifier = Modifier
+) {
+  var currentMonth by remember { mutableStateOf(Calendar.getInstance()) }
+  currentMonth.time = selectedDate
+
+  Column(modifier = modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
+    // Header with navigation arrows
+    Row(
+        modifier = Modifier.fillMaxWidth().padding(8.dp),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically) {
+          // Previous month arrow
+          IconButton(
+              onClick = {
+                currentMonth.add(Calendar.MONTH, -1)
+                onDateSelected(currentMonth.time)
+              }) {
+                Icon(
+                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = "Previous Month")
+              }
+
+          // Month and year text
+          BasicText(
+              text =
+                  "${
+                    currentMonth.getDisplayName(
+                        Calendar.MONTH, Calendar.LONG, Locale.getDefault()
+                    )
+                } ${currentMonth.get(Calendar.YEAR)}",
+              style = MaterialTheme.typography.bodyLarge)
+
+          // Next month arrow
+          IconButton(
+              onClick = {
+                currentMonth.add(Calendar.MONTH, 1)
+                onDateSelected(currentMonth.time)
+              }) {
+                Icon(
+                    imageVector = Icons.AutoMirrored.Filled.ArrowForward,
+                    contentDescription = "Next Month")
+              }
+        }
+
+    // Calendar grid
+    CustomCalendarView(
+        selectedDate = currentMonth.time,
+        events =
+            events.filter {
+              val eventCalendar = Calendar.getInstance().apply { time = it }
```

Comment:
It would have been better to move the filtering logic to Calendar ViewModel. Like this, it better follows the principles of the MVVM architecture.